.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+repos:
```

Comment:
Add a pre-commit configuration to the project

Reply:
More questions - does that update it locally (I presume so?)? Or do they need to do a `git pull`?
Will there ever be a circumstance where you do NOT want what the pre-commit gives you?
What if people do a quick edit on github.com? Does it work there?

Reply:
I'll remove it. It's a bit slower to commit, but it only runs the checks on the modified files, so it's usually just a second or two.

Users only get it if they have pre-commit installed (it's a Python package and can also be installed using various package manager). Additionally, they must run `pre-commit install` (one time only) in the SQLFluff clone directory.

You can bypass the checks by adding the `--no-verify` option to your `git commit` command.

Pre-commit only runs on the client side, not on github.com. As such, it is a supplement to whatever CI checks are in place, _not_ a replacement.

Yes, it updates the files locally. The initial commit will fail, and you can review the locally changed files, then `git add` and `git commit` again.

Reply:
Ah OK. Sounds great. But think we should do as separate PR and also add documentation to CONTRIBUTING as part of that explaining how to use and benefits of it!